docs/reference/builder.md

```diff
@@ -530,8 +530,10 @@ FROM extras:${CODE_VERSION}
 CMD  /code/run-extras
 ```
 
-To use the default value of an `ARG` declared before the first `FROM` use an
-`ARG` instruction without a value:
+An `ARG` declared before a `FROM` is outside of a build stage, so it
```

Comment:
> Maybe something like "An ARG instructions is only relevant for the next FROM instruction that comes after it in the Dockerfile,

That's not exactly correct. There are two cases:
* any `ARG` before the first `FROM` can be used in any `FROM` line
* any `ARG` within a build stage (after a `FROM`) can be used in that build stage

I can add more details to the `ARG` reference section as well (this stuff is under `FROM`) , but I think this line needs to be updated as well something along the lines of what I have.